Dockerfile

```diff
@@ -0,0 +1,53 @@
+# This Dockerfile is composed of two steps: the first one builds the release
+# binary, and then the binary is copied inside another, empty image.
+
+#################
+#  Build image  #
+#################
+
+FROM ubuntu:bionic AS build
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    ca-certificates \
+    curl \
+    build-essential \
+    pkg-config \
+    libsqlite3-dev \
+    libssl-dev
+
+# Install the currently pinned toolchain with rustup
+RUN curl https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init >/tmp/rustup-init && \
+    chmod +x /tmp/rustup-init && \
+    /tmp/rustup-init -y --no-modify-path --default-toolchain stable
+ENV PATH=/root/.cargo/bin:$PATH
+
+# Build the dependencies in a separate step to avoid rebuilding all of them
+# every time the source code changes. This takes advantage of Docker's layer
+# caching, and it works by copying the Cargo.{toml,lock} with dummy source code
+# and doing a full build with it.
+WORKDIR /tmp/source
+COPY Cargo.lock Cargo.toml /tmp/source/
```

Comment:
Build fails here, `Cargo.lock` doesn't exist.

Reply:
Versioned the lockfile.